src/js/components/modals/UninstallPackageModal.js

```diff
@@ -0,0 +1,197 @@
+    };
+
+    this.store_listeners = [
+      {
+        name: 'cosmosPackages',
+        events: ['uninstallError', 'uninstallSuccess'],
+        unmountWhen: function () {
+          return true;
+        },
+        listenAlways: true
+      }
+    ];
+
+    METHODS_TO_BIND.forEach((method) => {
+      this[method] = this[method].bind(this);
+    });
+  }
+
+  onCosmosPackagesStoreUninstallError(packageUninstallError) {
+    this.setState({packageUninstallError, pendingUninstallRequest: false});
+  }
+
+  onCosmosPackagesStoreUninstallSuccess() {
+    let {handleUninstallFinish} = this.props;
+    this.setState(
+      {
+        uninstallSuccess: true,
+        packageUninstallError: null,
+        pendingUninstallRequest: false
+      },
+      handleUninstallFinish
+    );
+  }
+
+  handleClose() {
+    this.setState({
+      uninstallSuccess: false,
+      packageUninstallError: null,
+      pendingUninstallRequest: false
+    });
+    this.props.onClose();
+  }
+
+  handleUninstallPackage() {
+    let {cosmosPackage} = this.props;
+    let appId;
+    let packageName;
+    let packageVersion;
+    if (cosmosPackage) {
+      appId = cosmosPackage.getAppId();
+      packageName = cosmosPackage.getName();
+      packageVersion = cosmosPackage.getCurrentVersion();
+    }
+
+    CosmosPackagesStore.uninstallPackage(
+      packageName,
+      packageVersion,
+      appId
+    );
+
+    this.setState({pendingUninstallRequest: true});
+  }
+
+  getEmptyNode() {
+    return <div />;
+  }
+
+  getErrorMessage() {
+    let {packageUninstallError} = this.state;
+    if (!packageUninstallError) {
+      return null;
+    }
+
+    return (
+      <CosmosErrorMessage
+        className="text-error-state text-overflow-break-word"
+        error={packageUninstallError}
+        header="" />
+    );
+  }
+
+  getPostUninstallNotes() {
+    let {cosmosPackage} = this.props;
+
+    if (!cosmosPackage) {
+      return this.getEmptyNode();
+    }
+
+    let name = cosmosPackage.getAppIdName();
+    let notes = cosmosPackage.getPostUninstallNotes();
+
+    return (
+      <div className="container-pod container-pod-short text-align-center">
+        <span className="text-success">
+          <IconCircleCheckmark />
+        </span>
+        <h3 className="short-top">{`${name} Uninstalled`}</h3>
+        <p className="small flush-bottom">
+          {notes}
+        </p>
+      </div>
+    );
+  }
+
+  getUninstallModalContent() {
+    let {props: {cosmosPackage}, state: {uninstallSuccess}} = this;
+
+    if (uninstallSuccess) {
+      return this.getPostUninstallNotes();
+    }
+
+    if (!cosmosPackage) {
+      return this.getEmptyNode();
+    }
+
+    return (
+      <div className="container-pod container-pod-short text-align-center">
+        <h3 className="flush-top">Are you sure?</h3>
+        <p>
+          {`${cosmosPackage.getAppIdName()} will be uninstalled from ${Config.productName}. All tasks belonging to this package will be killed.`}
```

Comment:
We don't use the 80 character rule for strings and regex's, since it quickly becomes hard to read and verify if everything is correct (where it does not when it is one string). The situation is opposite for code; the short lines makes it easier to understand what is going on. That is why the 80 character rule is not a hard rule.

We've ended up in a lot of bad situations with class names for example.